### DISCUSSION TITLE
Error publishing new pages

### DISCUSSION THREAD

**Discussion by Aridio** [givanz/Vvveb#145] [Q&A]
Hey guys! I'm not able to save any page, not all options appear on the page of creation page

<img width="1440" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/f4e9d3a2-1d23-41d7-8914-d873ce6af8bf">

When I save the new page this error page is shown:

<img width="1440" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/50fb5421-23d3-478d-b8d8-f0bcdae1514b">

I'll share the system info:

`General

      Vvveb version
      1.0.2

      Admin path
      /vvveb/admin/

      PHP version
      8.2.0 | apache2handler

      Server
      Apache/2.4.54 (Unix) OpenSSL/1.0.2u PHP/8.2.0 mod_wsgi/3.5 Python/2.7.18 mod_fastcgi/mod_fastcgi-SNAP-0910052141 mod_perl/2.0.11 Perl/v5.30.1

      OS version
      Darwin Aridios-MacBook-Pro.local 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:16:51 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8103 arm64

      Database driver & version
      mysqli | Client library version: mysqlnd 8.2.0 | Server version: 5.7.39

      PHP time limit
      30

      PHP memory limit
      128M

      Max input time
      60

      Upload max filesize
      32M

      PHP post max size
      8M

      Extensions
      Core date libxml openssl pcre sqlite3 zlib bcmath bz2 calendar ctype curl dom hash fileinfo filter ftp gd gettext json iconv intl SPL ldap mbstring session standard exif mysqlnd PDO pdo_mysql pdo_sqlite Phar posix random Reflection mysqli SimpleXML soap sockets sodium tokenizer xml xmlreader xmlwriter xsl zip apache2handler imap pgsql pdo_pgsql imagick

      Page cache
      enabled

      Object cache
      APCu

      Email Driver
      mail

      Session Driver
      php

      Debug
      enabled

      Sql changes check
      enabled

      Image library
      gd

Server

      Document root
      /Applications/MAMP/htdocs

      Public path
      /vvveb/public/

      Http host
      localhost:8888

      Http connection
      keep-alive

      Http sec ch ua
      "Not/A)Brand";v="8", "Chromium";v="126", "Google Chrome";v="126"

      Http sec ch ua mobile
      ?0

      Http sec ch ua platform
      "macOS"

      Http upgrade insecure requests
      1

      Http user agent
      Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36

      Http accept
      text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7

      Http sec fetch site
      same-origin

      Http sec fetch mode
      navigate

      Http sec fetch user
      ?1

      Http sec fetch dest
      document

      Http referer
      http://localhost:8888/vvveb/admin/index.php

      Http accept encoding
      gzip, deflate, br, zstd

      Http accept language
      en-US,en;q=0.9,es-DO;q=0.8,es;q=0.7

      Http cookie
      handl_original_ref=http%3A%2F%2Flocalhost%3A8888%2Fdermapointclinic%2Fwp-admin%2Fpost.php%3Fpost%3D24%26action%3Dedit; handl_ip=%3A%3A1; handl_landing_page=http%3A%2F%2Flocalhost%2Fdermapointclinic%2Fwp-json%2Fwp%2Fv2%2Fusers%3Fcontext%3Dview%26who%3Dauthors%26per_page%3D50%26_fields%3Did%252Cname%26_locale%3Duser; handl_ref=http%3A%2F%2Flocalhost%3A8888%2Fdermapointclinic%2Fwp-admin%2Fpost.php%3Fpost%3D24%26action%3Dedit; handl_url=http%3A%2F%2Flocalhost%2Fdermapointclinic%2Fwp-json%2F%3F_fields%3Ddescription%252Cgmt_offset%252Chome%252Cname%252Csite_icon%252Csite_icon_url%252Csite_logo%252Ctimezone_string%252Curl%26_locale%3Duser; PHPSESSID=37iub67i5bnu5mueqqshp9v44o; user=1

      Path
      /usr/bin:/bin:/usr/sbin:/sbin

      Server signature

      Server software
      Apache/2.4.54 (Unix) OpenSSL/1.0.2u PHP/8.2.0 mod_wsgi/3.5 Python/2.7.18 mod_fastcgi/mod_fastcgi-SNAP-0910052141 mod_perl/2.0.11 Perl/v5.30.1

      Server name
      localhost

      Server addr
      ::1

      Server port
      8888

      Remote addr
      ::1

      Request scheme
      http

      Context prefix

      Context document root
      /Applications/MAMP/htdocs

      Server admin
      you@example.com

      Script filename
      /Applications/MAMP/htdocs/vvveb/admin/index.php

      Remote port
      52750

      Gateway interface
      CGI/1.1

      Server protocol
      HTTP/1.1

      Request method
      GET

      Query string
      module=tools/systeminfo

      Request uri
      /vvveb/admin/index.php?module=tools/systeminfo

      Script name
      /vvveb/admin/index.php

      Php self
      /vvveb/admin/index.php

      Request time float

1719846081.244606

      Request time

1719846081`

And the error log file:
[php_error.log](https://github.com/user-attachments/files/16055301/php_error.log)

And just in case, my device info:
<img width="490" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/9dde9adc-21aa-41fd-a157-897ad31b609c">

To finish the MAMP Server version I'm using is 6.9.

I hope you can help, I really loved this project, thanks in advance!

**Comment by givanz** [comment]
Hi

At first glance it looks like there is no language configured, there must be at least a language configured (default).

Can you please share your language configuration from admin > settings > localization > languages?
https://demo.vvveb.com/admin/index.php?module=localization/languages

I will add a warning to prevent deleting all languages.

**Reply by Aridio** [reply]
Great, that solved the issue, but I still see some other issues like the following:

The added pages are visible only on incognito, I cleared cache on chrome but it doesn't matter
![incognit](https://github.com/givanz/Vvveb/assets/9813096/989b7a07-921e-499f-a897-399d1b71cec3)

By default the site URL is the host root instead of the subdirectory that I defined at the beginning, so I changed it on page settings. At the same time I had to change every url of the menu adding the subdirectory name "/vvveb/page/services" (I'm not sure if this is how it should work)
<img width="891" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/9a4e871d-613f-4b98-b185-9ffc72a4110f">

Other weird thing I saw is the autocomplete doesn't work on menu page when I adding a page link, if I type the page name and I save, on the live site, when I click on the added link it's linked to "http://localhost:8888/vvveb/public/themes/landing/url" instead of "http://localhost:8888/vvveb/page/services"
<img width="271" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/cb67485a-8dc9-470b-ba0b-8792590d10f3">

Also when I click on Pages -> Taxonomi the admin is corrupt and all links break, for example if I click on view live website the url opened is "http://localhost:8888/vvveb/public/admin/default/@@base_href#" instead of "http://localhost:8888/vvveb/", and if I click on Pages I got "http://localhost:8888/admin/?module=content/posts&type=page", the subdirectory that I configured at the beginning is lost, etc...
<img width="1440" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/03a60d40-c734-43f7-9d46-c00ed0fe5848">
 
I'll continue exploring the platform to see if I can find more issues, I hope these will help to improve further!

**Comment by givanz** [comment]
Thanks for the bug reports and testing

> The added pages are visible only on incognito, I cleared cache on chrome but it doesn't matter

In the first incognito screenshot the active site name is shown (Default) in the sites selector but in the second screenshot the name is not shown.

It looks like the active site was deleted and the session has the missing site as active.

You can select the new site from the dropdown and it should work. 

I added checks to prevent deleting all sites and set the next site as active when the active site is deleted.

> By default the site URL is the host root instead of the subdirectory that I defined at the beginning, 

I fixed the view site url to include the subdirectory path.

>  I had to change every url of the menu adding the subdirectory name

Menu entries in demo data were are added as simple link type instead of pages, I chanded the entries to page type to display properly for subdirectory installs.

> autocomplete doesn't work on menu page when I adding a page link

I made changes to autocomplete to work with partial name search, it was using full text search.

> when I click on Pages -> Taxonomi the admin is corrupt and all links break

Menu headers don't have a link, it's bug that they are clickable, I changed them so they can't be clicked.

The page with wrong links the is the admin theme loaded without any processing, this is shown because an empty link was clicked and the browser opens the base url by default.

---

To fix the issues please redownload `latest.zip` and overwrite the files, you might need to clear site and browser cache.

**Comment by Aridio** [comment]
Cool! I've tested, some things where solved but other not at all and other new appeared:

Publish date of a new page seems bugged, it does nothing when clicking:
<img width="274" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/e11aad1c-1878-4a9a-9001-2c818c2efa22">

Autocomplete on item menu creation seems not work yet:
<img width="350" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/f442b019-7ddd-4efa-b609-f545e59c99e8">

Default home link is still pointing to the root path instead of the defined subdirectory path:
<img width="663" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/4c9a669b-4c19-4107-b599-4a26086a6814">

> Menu entries in demo data were are added as simple link type instead of pages, I chanded the entries to page type to display properly for subdirectory installs.

I'm not sure if that change was applied correctly.

When I click on edit site appears this message:
<img width="712" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/632ac12d-c613-4463-adec-321a34a7627c">

Admin toolbar doesn't appearing on default home page:
<img width="1440" alt="image" src="https://github.com/givanz/Vvveb/assets/9813096/24649c92-75d2-4255-80cb-a278b1d03e36">

If I can find more things I'll let you know!

You guys are making a great job, congrats! 👌

**Comment by givanz** [comment]
> Publish date of a new page seems bugged, it does nothing when clicking:

It does not show the date time picker on input focus? 

Can you please check developer tools (F12 in chrome) for error messages?

Does it work on demo to rule out browser specific causes? https://demo.vvveb.com/admin/index.php?module=content/post&post_id=7&type=page

> Autocomplete on item menu creation seems not work yet:

Is the default site with demo pages deleted and a new site without pages selected?

Can you please try clearing browser cache? maybe the old js code is loaded. 

I changed autocomplete to show a list of pages on focus even before typing.

Can you also check developer tools for js errors?

> Default home link is still pointing to the root path instead of the defined subdirectory path:

I added code to update all menu links on install when using subdir installation, to reinstall just delete `config/db.php` and install page will show.

> When I click on edit site appears this message:

I fixed the bug in vtpl.php

> Admin toolbar doesn't appearing on default home page:

This usually happens when different subdomains are used for admin and frontend, a common case is when one is accessed from www.domain.com and another from [domain.com](domain.com)

This causes different sessions to be used on each subdomain, this is easily solved by enabling `V_SHARED_SESSION` https://github.com/givanz/Vvveb/blob/master/env.php#L50 

But on localhost I'm not sure what can cause this.

> If I can find more things I'll let you know!

> You guys are making a great job, congrats! 👌

Thank you for testing and reporting bugs and appreciation.
